jaxrs-api/src/main/java/jakarta/ws/rs/core/Response.java

```diff
@@ -1131,6 +1136,13 @@ public ResponseBuilder status(final Status status) {
         public default Status toEnum() {
             return Status.fromStatusCode(getStatusCode());
         }
+
+        /**
+         * @deprecated May return false when reasonPhrase differs. Instead compare {@code statusCode} directly.
+         */
+        @Deprecated
+        @Override
+        abstract boolean equals(Object obj);
```

Comment:
I'm still mulling this over, but I think I don't like deprecating the `equals` method.  I like the clarification that users should not depend on `equals` in cases where the status code is the same, but the reason code might be different, but maybe instead of deprecating the `equals` method we could specify that an implementation must check statusCode, reasonPhrase.  Something like:
```
/**
 * Returns true only if this passed-in object is of {@code StatusType}, is non-null and contains the same
 * {@code statusCode}, {@code reasonPhrase}, and {@code family} as this {@code StatusType}.
 *
 * @param obj - the reference object with which to compare.
 * @return true if this object is the same as the obj argument; false otherwise.
 * @since 3.1
 */
abstract boolean equals(Object obj)
```

Wdyt?

Reply:
Good point. Yeah, thinking about this more, exactly specifying how `equals()` should behave may be better than deprecating it. So +1 for your suggestion from my side.

Reply:
I don't like deprecating the method `equals` method either. However, I think there's still an issue if an operand is an instance of a class that implements `StatusType` and the other is one of the enum values, even if all the fields match. It would depends on which operand is left vs. right in `equals`. So I think it is possible for the condition as described in the newly proposed Javadoc above to be true and for the method to return false. I'd probably just add a clarification to `getStatusInfo`.

Reply:
I second that. We should not deprecate the equals method.